script.js

```diff
@@ -1,12 +1,355 @@
+    selectE.style.display = 'none';
+    //home button
+    makeHomeBtn()
+  })
+
+  return header
+}
+
+//section 
+function createSection() {
+  let section = document.createElement("div");
+  section.className = "section";
+  let link = document.createElement("div");
+  link.className = 'maze';
+  link.innerHTML = `Originally, data was obtained from <a href = 'https://www.tvmaze.com/' target= '_blank'>TVMaze.com</a>`
+  section.appendChild(link);
+  return section;
+}
+
+//popup episode at selection
+let episodes = []; // global variable
+
+function createPopUp(episodeId) {
+  episodeId = parseInt(episodeId)
+  const popUp = document.createElement("div");
+  popUp.id = "popupEpisode";
+  popUp.className = "episodeItem";
+
+  // 1. find the episode with this id
+  const allEpisodes = episodes;
+  console.log(episodeId, allEpisodes);
+  const episodeData = allEpisodes.find((episode) => episode.id === episodeId);
+
+  popUp.innerHTML = `
+  <button id="popupCloseBtn">X</button>
+  <h2>${episodeData.name} - ${pad(episodeData)}</h2>
+  <div class="image">
+    <img src=${episodeData.image.medium} alt='${episodeData.season} ${episodeData.name
+    }'/>
+  </div>
+  <div class="description">
+    ${episodeData.summary}
+  </div>
+`;
+
+  // button for closing the popUp:
+  //get the button and add event listener
+  let closeBtn = popUp.children[0];
+  closeBtn.addEventListener("click", (e) => {
+    // remove the popup from the DOM
+    popUp.remove();
+  });
+
+  return popUp;
+}
+
+
+function makePageLayout() {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  //header
+  let header = createHeader();
+  rootElem.appendChild(header);
+  //section
+  let section = createSection();
+  // section will contain episodes, so it is hidden at the beginning
+  section.style.display = 'none';
+  rootElem.appendChild(section);
+  let selectE = document.querySelector(".select");
+  selectE.style.display = 'none';
+  // the popup will show the chosen episode.
+  // when selected value changes, a popup with the episode will open
+  selectE.addEventListener("change", (e) => {
+    let episodeId = e.target.value;
+    const popUp = createPopUp(episodeId);
+    rootElem.appendChild(popUp);
+  });
+
+
+}
+
+function makeHomeBtn() {
+  let section = document.querySelector('.section')
+  let showView = document.querySelector('.showView')
+
+  if (section.style.display == 'none') {
+    section.style.display = '';
+  } else {
+    section.style.display = 'none';
+  }
+
+  if (showView.style.display == 'none') {
+    showView.style.display = '';
+  } else {
+    showView.style.display = 'none';
+
+  }
+}
+
+function hideShowView() {
+  let section = document.querySelector('.section')
+  let showView = document.querySelector('.showView')
+  section.style.display = '';
+
+  showView.style.display = 'none';
+
+}
+
+//to display page
+function makePageForEpisodes(episodeList) {
```

Comment:
Amazing work. Great understanding on functions especially writing code and programs that are in blocks and easy to understand and improve. Reference to line 213 to 268